### PR TITLE
polaris-web: use buildNpmPackage

### PR DESCRIPTION
**Pull request by dotlambda**
###### Description of changes
See https://github.com/NixOS/nixpkgs/issues/229475.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by natsukium — COMMENTED** [review]
We can drop `meta.platforms` now that we use buildNpmPackages.

File: pkgs/servers/polaris/web.nix [hunk 1]
```diff
@@ -1,65 +1,31 @@
...
-    sha256 = "2XqU4sExF7Or7RxpOK2XU9APtBujfPhM/VkOLKVDvF4=";
+    rev = "build-${version}";
+    hash = "sha256-mhrgHNbqxLhhLWP4eu1A3ytrx9Q3X0EESL2LuTfgsBE=";
   };
 
+  npmDepsHash = "sha256-lScXbxkJiRq5LLFkoz5oZsmKz8I/t1rZJVonfct9r+0=";
+
   nativeBuildInputs = [
-    nodejs
+    nodePackages.node-gyp-build
```
Comment (dotlambda) [inline pkgs/servers/polaris/web.nix:22]: @lilyinstarlight Is this the way to go despite using `buildNpmPackage`?

Comment (SuperSandro2000) [reply pkgs/servers/polaris/web.nix:22]: I think we should probably package node-gyp-build separately so that we do not rely on nodePackages anymore

Comment (lilyinstarlight) [reply pkgs/servers/polaris/web.nix:22]: If the package depends on `node-gyp-build`, it should be in the `devDependencies` in `package.json` (but `node-gyp` itself is included in the nodejs distribution and should not be included)

Does it not do that?

Comment (lilyinstarlight) [reply pkgs/servers/polaris/web.nix:22]: Oh I admittedly had to look up what node-gyp-build was again, but this is the thing that basically checks whether prebuilt binaries exist in the downloaded package and runs its own `node-gyp rebuild` if not

I'm wondering if we should just make a fake `node-gyp-build` command that just does `node-gyp rebuild`...

**Comment by dotlambda** [comment]
> We can drop `meta.platforms` now that we use buildNpmPackages.

done

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by pbsds** [comment]
#203146?

**Review by pbsds — APPROVED** [review]

**Pull request merged**